# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/broadcastchannel/BroadcastChannel.cpp
# Commit: b1ab6c750d04
# Full Hash: b1ab6c750d04299011c1446b1752c2ff98a92b33
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1800659
# File Overlap Count: 1
# Description:
#   Bug 1800659 - P3 Remove WorkerPrivate::ClearMainEventQueue() r=asuth,smaug
#   
#   Pending->Canceling->Killing (WorkerNeverRan)
#   
#   Need to clear WorkerPrivate::mPreStartRunnables
# ==============================================================================

diff -r 32515f9b41c7 -r b1ab6c750d04 dom/broadcastchannel/BroadcastChannel.cpp
--- a/dom/broadcastchannel/BroadcastChannel.cpp	Tue Jun 06 06:36:50 2023 +0000
+++ b/dom/broadcastchannel/BroadcastChannel.cpp	Tue Jun 06 06:36:50 2023 +0000
@@ -279,6 +279,10 @@
     agentClusterId = global->GetAgentClusterId();
   }
 
+  if (!global->IsEligibleForMessaging()) {
+    return;
+  }
+
   RefPtr<SharedMessageBody> data = new SharedMessageBody(
       StructuredCloneHolder::TransferringNotSupported, agentClusterId);
 